notification/check/threshold.go

```diff
@@ -47,21 +48,145 @@ func (c Threshold) MarshalJSON() ([]byte, error) {
 		})
 }
 
+type thresholdDecode struct {
+	Base
+	Thresholds []thresholdConfigDecode `json:"thresholds"`
+}
+
+type thresholdConfigDecode struct {
```

Comment:
I'm not sure I follow this pattern, why not do something like https://github.com/influxdata/influxdb/blob/4172dfd09eeca6fc7ca11696feb5c46f3206dcb7/dashboard.go#L316

Reply:
this one is much simpler than dashboard, by doing this we only need to do one unmarshaling.

Reply:
It's simpler, but it feels a little sloppy to me. In any case though, it's a minor thing. Feel free to ignore.